lib/atlas/active_document/share_group_inclusion_validator.rb

```diff
@@ -0,0 +1,21 @@
+module Atlas
+  module ActiveDocument
+    class ShareGroupInclusionValidator < ShareGroupValidator
+      def validate(record)
+        return if record.errors.messages[options[:attribute]]
+
+        share_groups_for(record).each_pair do |share_group, inputs|
+          missing = input_class.by_share_group[share_group] - inputs.keys
```

Comment:
Just a quick note but `share_group` is not available outside of the loop . 🤔 I'm looking how to cache `input_class.by_share_group` though 😄